python/mlc_chat/serve/conversation.py

```diff
@@ -0,0 +1,126 @@
+"""The conversation templates in MLC LLM serving"""
+
+from typing import Dict, List, Optional, Tuple
+
+from pydantic import BaseModel, Field, field_validator
+
+# The system message placeholder in the system message prompt.
+SYSTEM_MESSAGE_PLACEHOLDER = "<<SYSTEM MESSAGE PLACEHOLDER>>"
+
+
+class Conversation(BaseModel):
```

Comment:
This is not related to the conversation template defined in our previous C++ codebase, is that correct? I wonder can we unify them?

Reply:
I think it would be good to have a common json schema(as being specified in here via pydantic), then embed all conv template in json. So we can only keep a c++ impl of the conv construction, all relies on a common spec.

The model here allows the API endpoint to also pass in conv template in future(this is a super set of openai api)